### DISCUSSION TITLE
Can screenshots taken by snap feature of tiddlyclip be embedded as external file using canonical uri instead of saving as an image tiddler

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#51] [Q&A] [answered]
[File upload plug in ](https://talk.tiddlywiki.org/t/aug-17th-2021-community-call-core-savers-syncers-upload-plugin-discourse-forum/28)helps to import images into the wiki as external files by uploading to external or local storage. I have tried the WebDAV server option of the file upload plug in and its easy to use. Can the screenshots taken by snap feature of tiddlyclip be embedded as external file using canonical uri like the file upload plug in instead of saving as an image tiddler

**Comment by buggyj** [comment, marked as answer]
Yes but an addition plugin 'media saver' needs installing in firefox - 

instructions to install the addon are here

https://github.com/buggyj/savemedia

The images are store in a subdirectory call 'media' within the directory where the tiddlywiki is. So to store snaps separately put each tiddlywiki in a separate directory.

And the rule needs changing.

![image](https://user-images.githubusercontent.com/2855795/149411374-ae5c4901-f309-4761-a0fd-2d4c20c381d4.png)

i.e.

`|null||||{"#ext":"png"},{"%title":"((*@pageTitle*))((*@dateTimeShort*)).((*#ext*))"}|nosave|`
`|((*@pageTitle*))/thumbprint|[img[((*@csaver(%title,@snap)*))]]|ClippedImage||{"$when":"((*@dateTimeShort*))"},{"$location":"((*@pageRef*))"}|[[SnapRuleMode]]|`
`|((*@pageTitle*))|((*@pageRef*))\n\n{{((*@pageTitle*))/thumbprint}}\n\n((*@text*))\n\n|||{"$location":"((*@pageRef*))"},{"$caption":"((*@pageTitle*))"},{"$when":"((*@dateTimeShort*))"}|append add|`

**Reply by arunnbabu81** [reply]
It seem to be working in my short test. I didn’t get enough time today to test it in detail though. Was little busy at my job. Will give feedback after detailed testing soon. Thanks for all the help.

**Reply by arunnbabu81** [reply]
Local saving of snaps are working. But I had some trouble making it to work when I was using WebDAV server, So I changed to your [savetiddlers](https://github.com/buggyj/savetiddlers) add-on

**Reply by arunnbabu81** [reply]
Currently the snaps saved locally is shown within the tiddler in this format [img[media/filename.png]]

Is there any way to show it in this format <<lbox "media/filename.png">>

This is because of i want to make use of the [light box macro](https://kookma-lightbox.tiddlyhost.com/#)

**Reply by buggyj** [reply]
you need to change

`[img[((*@csaver(%title,@snap)*))]]`

to

`<<lbox "((*@csaver(%title,@snap)*))">>`

**Reply by arunnbabu81** [reply]
I made new rule like this
![image](https://user-images.githubusercontent.com/67494083/150666781-866bd45c-70d0-41f9-97ee-106f353eb520.png)

![image](https://user-images.githubusercontent.com/67494083/150666795-327ce2f0-e463-4eb8-827e-3d0c3dea6e3b.png)

When i take snap, i get a notification like this

![image](https://user-images.githubusercontent.com/67494083/150666750-3e56cf2c-7b42-4017-b909-fca73113f2f2.png)

and the snap looks like this

![image](https://user-images.githubusercontent.com/67494083/150666803-ef33ba5e-1031-4d41-a7c4-96024ecbe1b8.png)

**Reply by buggyj** [reply]
my bad. The double quotes need to be 'escaped' i.e.

`<<lbox \"((*@csaver(%title,@snap)*))\">>`

or use single quotes

`<<lbox '((*@csaver(%title,@snap)*))'>>`

**Reply by arunnbabu81** [reply]
Great!! That was awesome

**Reply by arunnbabu81** [reply]
One more doubt....i tried to use "userstring" for page title for snaps and it was not working. 

This is what i tried

![image](https://user-images.githubusercontent.com/67494083/150690937-95129774-41ee-46ee-808e-150fa6f8b18a.png)

![image](https://user-images.githubusercontent.com/67494083/150690969-7edcd900-3da4-4d43-a4a2-6b44651d6e84.png)

**Reply by buggyj** [reply]
'userstring' must be adding to the row in the TiddlyClipConfig:

![image](https://user-images.githubusercontent.com/2855795/150692721-baae5cb7-25e0-4763-8f2c-80e065a179f1.png)

**Reply by arunnbabu81** [reply]
is it possible to use "userstring" as title for the locally saved snap

**Reply by buggyj** [reply]
the file name is created on the first line in your rule in the variable '%title' from @pageTile , you could replace this with @userstring, but, some characters cannot be used in file names (eg '/') so this may not always work well.

**Comment by arunnbabu81** [comment]
> you need to change
> 
> `[img[((*@csaver(%title,@snap)*))]]`
> 
> to
> 
> `<<lbox "((*@csaver(%title,@snap)*))">>`

How to modify the defaultPin to make it work with lightbox macro as shown in the quoted reply

![image](https://user-images.githubusercontent.com/67494083/151235496-2ab8c6c1-2535-43cd-aaba-42b118f57222.png)

**Reply by buggyj** [reply]
change

`[img[((*@onImage*??*@imageURL*??*@largestImgURL*))]]`

to 

`<<lbox '((*@onImage*??*@imageURL*??*@largestImgURL*))'>>`
